### DISCUSSION TITLE
fiters definition in web.xml

### DISCUSSION THREAD

**Discussion by b-noam** [quarkiverse/quarkus-cxf#687] [Q&A]
we are migrating code from legacy j2ee app where we used filters in web.xml. these filters catch REST/SOAP/WEB request and did some logic. 
in Quarkus filters works well for REST but for SOAP they are not applied and request goes directly to the service 

any idea how we can make it work

**Comment by shumonsharif** [comment]
> in Quarkus filters works well for REST but for SOAP they are not applied and request goes directly to the service

Hi @b-noam How are you implementing your Quarkus filters; and are you able to provide a reproducer, or at minimum some sample code?  I believe we've successfully addressed a few issues in the past using `@RouteFilter` ... so please keep us posted on how you're going about it, as there are many ways to implement filters.

**Comment by b-noam** [comment]
we are using filters that implements javax.servlet.Filter

In web.xml file we have definition  like:

```xml
<filter>
   <filter-name>myFilter</filter-name>
   <filter-class>com.xyz.MyFilter </filter-class>
</filter>
```

And filter mapping to apply the filter on a path

i.e. for REST we use

```xml
<filter-mapping>
    <filter-name>myFilter</filter-name>
    <url-pattern>/REST/services/*</url-pattern>
</filter-mapping>
```

And for SOAP we use

```xml
<filter-mapping>
    <filter-name>myFilter</filter-name>
    <url-pattern>/SOAP/services/*</url-pattern>
</filter-mapping>
```

We have end point defined for REST/SOAP on the given path – REST calls are handled by the filter but SOAP calls are not going through the filters

**Comment by shumonsharif** [comment]
Hi @b-noam It's not clear to me, but I believe you are currently able to use your filters as-is is because you are using the `quarkus-undertow` libraries, which provide access to the Servlet APIs (please correct me if I'm mistaken).

I'm afraid the `quarkus-cxf` extension currently directly uses the Vert-X transport layer in Quarkus, and it is not integrated with `quarkus-undertow`.  I believe your best option may be to implement a `@RouteFilter` (as described [here](https://quarkus.io/guides/reactive-routes#intercepting-http-requests)) and port your filter logic.

@famod and @ppalaga may have some ideas on this as well, so tagging them.

**Reply by ppalaga** [reply]
> I'm afraid the `quarkus-cxf` extension currently directly uses the Vert-X transport layer in Quarkus, and it is not integrated with `quarkus-undertow`. I believe your best option may be to implement a `@RouteFilter` (as described [here](https://quarkus.io/guides/reactive-routes#intercepting-http-requests)) and port your filter logic.

Nothing to add. Indeed quarkus-cxf is not servlet-based and you'll have to use whatever CXF or vert.x means to reach what you need.

**Comment by b-noam** [comment]
Thanks for the quick response – you did got it right
We will look into your suggestion.t
